### PR TITLE
Allow subdirectories for add & delete granules

### PR DESCRIPTION
**Pull request by ritvje**
Add a config option that allows keeping the subpath given in arguments when adding granules. 

E.g. if config file has

```yaml
geoserver_target_dir: /smartmet/radar/geotiff/radar-optclass/
```
if the filepath for `add_granule.py` is given as `fivih_3067/20230926_1300_fivih_ppi-0.3_optclass.tif`, the filepath is interpreted as 

`/smartmet/radar/geotiff/radar-optclass/fivih_3067/20230926_1300_fivih_ppi-0.3_optclass.tif`

### CODE REVIEW

**Review by pnuu — CHANGES_REQUESTED** [review]
This seems to include also the changes in #19 . Please include only the changes of the new feature so that the changes still needed for #19 are not needed to be added also in this one.

File: georest/__init__.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def add_file_to_mosaic(config, fname_in, filesystem='posix'):
     This function wraps some boilerplate around adding a granule to a layer.
 
     """
-    fname = utils.convert_file_path(config, fname_in)
+    fname = utils.convert_file_path(config, fname_in, keep_subpath=config["keep_subpath"])
```
Comment (lahtinep) [inline georest/__init__.py:289]: Could also be
```suggestion
    fname = utils.convert_file_path(config, fname_in, keep_subpath=config.get("keep_subpath", False))
```
so forgetting the extra option from the config doesn't cause a crash.

File: georest/__init__.py [hunk 2]
```diff
@@ -397,15 +397,14 @@ def _delete_id_from_gs(cat, workspace, store, store_obj, granule):
...
-                fname, workspace, store)
+    logger.info("Granule '%s' removed from %s:%s", fname, workspace, store)
 
 
 def _delete_files_from_fs(config, gs_location):
     delete_files = config.get("delete_files", False)
     if not delete_files:
         return
-    fs_path = utils.convert_file_path(config, gs_location, inverse=True)
+    fs_path = utils.convert_file_path(config, gs_location, inverse=True, keep_subpath=config["keep_subpath"])
```
Comment (lahtinep) [inline georest/__init__.py:407]:
```suggestion
    fs_path = utils.convert_file_path(config, gs_location, inverse=True, keep_subpath=config.get("keep_subpath", False))
```

File: georest/utils.py [hunk 3]
```diff
@@ -85,9 +85,12 @@ def _write_property_zip(prop_paths, zip_path):
     return zip_path
 
 
-def convert_file_path(config, file_path, inverse=False):
+def convert_file_path(config, file_path, inverse=False, keep_subpath=False):
```
Comment (lahtinep) [inline georest/utils.py:88]: Apparently I haven't had a test case for this function, but we could add something now to `test_utils.py`. I'll write some and push them to this branch in a moment.

**Review by lahtinep — COMMENTED** [review]
Just a couple of comments that would simplify the configs a bit for cases where sub directories don't need to be handled.

I'll also push some additional tests for `georest.utils.convert_file_path()` to this branch in a moment when I've finalized them.

**Comment by lahtinep** [comment]
@ritvje do you think the cases I added for `keep_subpath=True` tests for `georest.utils.convert_file_path()` make sense?

**Comment by ritvje** [comment]
Looks good. Your version of the inverse option of `georest.utils.convert_file_path()` is much better, I was struggling a bit with that.

**Review by lahtinep — APPROVED** [review]
Looking good, merging.

**Pull request merged**